lib/error_tracker/schemas/occurrence.ex

```diff
@@ -6,10 +6,21 @@ defmodule ErrorTracker.Occurrence do
   in which the exception raised.
   """
 
+  import Ecto.Changeset
+
   use Ecto.Schema
 
   require Logger
-  import Ecto.Changeset
+
+  @type t :: %__MODULE__{
+          id: non_neg_integer(),
+          context: map(),
+          reason: String.t(),
+          stacktrace: map(),
+          error_id: non_neg_integer(),
+          error: ErrorTracker.Error.t(),
+          inserted_at: DateTime.t()
+        }
```

Comment:
We don't need that much detail for now. It only complicates things. For example if we go through this path we would also need to typespect the `Stacktrace` and its dependencies as well.

Let's keep it simple and do the same in every other schema:

```suggestion
  @type t :: %__MODULE__{}
```